### PR TITLE
Refactor calculator component

### PR DESCRIPTION
**Pull request by maen-al-khraisha**
Using hooks instead of class:
- modify setState to useState.

### CODE REVIEW

File: src/components/calculator/calculator.js [hunk 1]
```diff
@@ -1,197 +1,195 @@
...
-    this.setState(newState, () => {
-      let { total, operation, next } = this.state;
-      total = total || '';
-      operation = operation || '';
-      next = next || '';
-      let result = total + operation + next;
-      result = result === '' ? 0 : result;
+    const newExpresion = calculate(expresion, btnName);
+    setExpresion(newExpresion, () => {
+      const result = prepareData(newExpresion);
```
Comment (Kingstalux) [inline src/components/calculator/calculator.js]: - [x] Good job so far but your events are not working. Kindly fix this console error and make sure your events work as expected 👍 
- Your app:
Screenshot 2022-06-30 114954

**Review by Kingstalux — CHANGES_REQUESTED** [review]
Hi @maenkhraisha, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights 👏 
✅ Linters are passing 👌 
✅ Professional README file 💯 
✅ Correct Gitflow used 👍 


## Required Changes ♻️
_Check the comments under the review._
- [x] Kindly provide a PR description which provides information on all the changes you made in this PR 😄 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/calculator/calculator.js (hunk 1 above)
Comment (maen-al-khraisha) [reply src/components/calculator/calculator.js]: done 👍 
Thank you 😃

**Review by aimemalaika — APPROVED** [review]
Hi @maenkhraisha, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

133591850-fdcb7dd3-dd7a-4274-81e7-0e59434148f7

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**